packages/runtime/container-runtime/src/compatUtils.ts

```diff
@@ -0,0 +1,164 @@
+		| "summaryOptions"
+	>
+>;
+
+/**
+ * Mapping of DocSchemaAffectingRuntimeOptions to their compatibility related configs.
+ *
+ * Each key in this map corresponds to a property in DocSchemaAffectingRuntimeOptions.
+ * The value is an object that maps a version string to the default value for that property when using compatibilityMode of at least that version.
+ *
+ * For example if the compatibilityMode is "1.5.0", then the default value for `enableGroupedBatching` will be false. If the compatibilityMode is "2.0.0"
+ * or later,then the default value for `enableGroupedBatching` will be true.
+ */
+const docSchemaAffectingOptionsConfigMap: ConfigMap<DocSchemaAffectingRuntimeOptions> = {
+	enableGroupedBatching: {
+		"1.0.0": false,
+		"2.0.0": true,
+	},
+	compressionOptions: {
+		"1.0.0": disabledCompressionConfig,
+		"2.0.0": enabledCompressionConfig,
+	},
+	enableRuntimeIdCompressor: {
+		// For IdCompressorMode, `undefined` represents a logical state (off). However, to satisfy the Required<>
+		// constraint we need to have it defined, so we trick the type checker here.
+		"1.0.0": undefined as unknown as "on" | "delayed",
+		// We do not yet want to enable idCompressor by default since it will increase bundle sizes,
+		// and not all customers will benefit from it. Therefore, we will require customers to explicitly
+		// enable it. We are keeping it as a DocSchema affecting option for now as this may change in the future.
+	},
+	explicitSchemaControl: {
+		"1.0.0": false,
+		// This option's intention is to prevent 1.x clients from joining sessions when enabled. Ideally, we would set this to true
+		// when the compatibility mode is set to >=2.0.0. However, this option is unique because it was not enabled by default prior
+		// to the implementation of compatibilityMode. Because `defaultCompatibilityMode` is set to "2.0.0", we need to ensure this option
+		// is not suddenly enabled by default unexpectedly. Therefore, we will set it to true starting at compatibilityMode>="2.0.1" to
+		// ensure that customers who do not provide compatibilityMode will not see any behavior change.
+		"2.0.1": true,
+	},
+	flushMode: {
+		// Note: 1.x clients are compatible with TurnBased flushing, but here we elect to remain on Immediate flush mode
+		// as a work-around for inability to send batches larger than 1Mb. Immediate flushing keeps batches smaller as
+		// fewer messages will be included per flush.
+		"1.0.0": FlushMode.Immediate,
+		"2.0.0": FlushMode.TurnBased,
+	},
+	gcOptions: {
+		"1.0.0": {},
+		// Explicitly disable running Sweep in compat mode "2". Although sweep is supported in 2.x, it is disabled by default.
+		// This setting explicitly disables it to be extra safe.
+		"3.0.0": { enableGCSweep: true },
+	},
+};
+
+/**
+ * Returns the default DocSchemaAffectingRuntimeOptions configuration for a given compatibility mode.
+ */
+export function getConfigsForCompatMode<
+	T extends Record<string, unknown> = DocSchemaAffectingRuntimeOptions,
```

Comment:
Where is this generic type param used besides with the default value?

Reply:
It's only for unit tests really